climada/engine/impact_calc.py

```diff
@@ -112,6 +112,22 @@ def impact(self, save_mat=True, assign_centroids=True,
         apply_deductible_to_mat : apply deductible to impact matrix
         apply_cover_to_mat : apply cover to impact matrix
         """
+        # check for compability of exposures and hazard type
+        if all(name not in self.exposures.gdf.columns for
+               name in ['if_', f'if_{self.hazard.haz_type}',
+                        'impf_', f'impf_{self.hazard.haz_type}']):
+            raise AttributeError(
+                "Impact calculation not possible. No impact functions found "
+                f"for hazard type {self.hazard.haz_type} in exposures."
+                )
```

Comment:
I think I would remove this test as it is more a test about the exposure in itself (i.e., it does more or less what `exposures.check()` does). The next test however is great as it looks at whether the impact computation can be performed.